integration-tests/src/test/java/tachyon/client/keyvalue/KeyValueStoresIntegrationTest.java

```diff
@@ -261,4 +268,146 @@ public void putKeyValueTooLargeTest() throws Exception {
+   * @return the URI to the store
+   * @throws Exception if any error happens
+   */
+  private TachyonURI createStoreOfSize(int size, List<KeyValuePair> pairs) throws Exception {
+    TachyonURI path = new TachyonURI(PathUtils.uniqPath());
+    KeyValueStoreWriter writer = sKVStores.create(path);
+    for (int i = 0; i < size; i ++) {
+      byte[] key = genBaseKey(i).getBytes();
+      byte[] value = genBaseValue(i).getBytes();
+      writer.put(key, value);
+      if (pairs != null) {
+        pairs.add(new KeyValuePair(key, value));
+      }
+    }
+    writer.close();
+
+    Assert.assertEquals(size, sKVStores.open(path).size());
+
+    return path;
+  }
+
+  private int getPartitionNumber(TachyonURI storeUri) throws Exception {
+    KeyValueMasterClient client =
+        new KeyValueMasterClient(ClientContext.getMasterAddress(), ClientContext.getConf());
+    return client.getPartitionInfo(storeUri).size();
+  }
+
+  /**
+   * Creates a store with the specified number of partitions.
+   *
+   * NOTE: calling this method will set {@link Constants#KEY_VALUE_PARTITION_SIZE_BYTES_MAX} to
+   * {@link Constants#MB}.
+   *
+   * @param partitionNumber the number of partitions
+   * @param keyValuePairs the key-value pairs in the store, null if you don't want to know them
+   * @return the URI to the created store
+   */
+  private TachyonURI createStoreOfMultiplePartitions(int partitionNumber,
+      List<KeyValuePair> keyValuePairs) throws Exception {
+    // These sizes are carefully selected, one partition holds only one key-value pair.
+    final long maxPartitionSize = Constants.MB; // Each partition is at most 1 MB
+    ClientContext.getConf().set(Constants.KEY_VALUE_PARTITION_SIZE_BYTES_MAX,
+        String.valueOf(maxPartitionSize));
+    final int keyLength = 4; // 4Byte key
+    final int valueLength = 500 * Constants.KB; // 500KB value
+
+    TachyonURI storeUri = new TachyonURI(PathUtils.uniqPath());
+    mWriter = sKVStores.create(storeUri);
+    for (int i = 0; i < partitionNumber; i ++) {
+      byte[] key = BufferUtils.getIncreasingByteArray(i, keyLength);
+      byte[] value = BufferUtils.getIncreasingByteArray(i, valueLength);
+      mWriter.put(key, value);
+      if (keyValuePairs != null) {
+        keyValuePairs.add(new KeyValuePair(key, value));
+      }
+    }
+    mWriter.close();
+
+    Assert.assertEquals(partitionNumber, getPartitionNumber(storeUri));
+
+    return storeUri;
+  }
+
+  /**
+   * Tests that a store of vairous sizes(including empty store) can be correctly deleted.
+   */
+  @Test
+  public void deleteStoreTest() throws Exception {
+    List<TachyonURI> storeUris = Lists.newArrayList();
+    storeUris.add(createStoreOfSize(0, null));
+    storeUris.add(createStoreOfSize(2, null));
+    storeUris.add(createStoreOfMultiplePartitions(3, null));
+
+    for (TachyonURI storeUri : storeUris) {
+      sKVStores.delete(storeUri);
+
+      mThrown.expect(TachyonException.class);
```

Comment:
in junit, once you have one `mThrown.expect` succeed, the test ends.
so you may want to do `try`-`catch` and assert failure if you don't really cach `TachyonException` for this particular test case